### DISCUSSION TITLE
Couldn't launch the NWB explorer

### DISCUSSION THREAD

**Discussion by memartone** [dandi/dandi-archive#2061] [General]
I tried to launch the NWB explorer from this dataset using the "Open with NWB Explorer" link but it never opened and just kept spinning for about an hour. https://dandiarchive.org/dandiset/000933/0.240315.1758/files?location=sub-1295011705&page=1

**Comment by yarikoptic** [comment]
"forwarded" upstream
- https://github.com/MetaCell/nwb-explorer/issues/340
but keeping this one open happen we need to address on our end (e.g. switch to use different URL etc)

**Comment by kabilar** [comment]
Hi @bendichter, do you know if NWB Explorer is still maintained?  It looks like the repo was last updated ~10 months ago.

**Comment by yarikoptic** [comment]
FWIW -- I think it is maintained as there already being feedback on that issue I have filed.

**Comment by kabilar** [comment]
Thanks, Yarik.

**Comment by pgleeson** [comment]
Hi @kabilar. Yes, NWB Explorer is still very much a core part of the https://www.opensourcebrain.org platform! We're in between funding cycles at the moment and just focussed on maintaining the current OSB platform, but not doing too much new feature development until a new grant starts. We're still very keen to hear of any issues you have with the current infrastructure.

**Comment by yarikoptic** [comment]
> We're still very keen to hear of any issues you have with the current infrastructure.

after that issue gets addressed, let's add some rule into our `upptime` checks like

https://github.com/dandi/upptime/blob/master/.upptimerc.yml#L48

to periodically test if services still integrate with dandi reliably.

**Comment by sanjayankur31** [comment]
Hi folks,

If you're testing for uptime, please use this URL:

https://nwbexplorer.v2.opensourcebrain.org/hub/home

Using a URL with a data file for NWBE will spawn anonymous pods on our google cloud cluster, and if they're not cleaned up in time, the cluster will hit its max limits for anonymous pods and not be able to create new ones. We're  putting a culling process in place that will kill "idle" anonymous pods after a while, but this will take some time to test and deploy.

**Comment by yarikoptic** [comment]
thank you @sanjayankur31 !  Could you please detail a little more -- ie. how to provide data file for this URL?  I see the page offering to start my own server and not a form to pass a file url

![image](https://github.com/user-attachments/assets/25a5ccaf-84bb-4b4b-8d7a-ac32aca7ba75)

**Comment by sanjayankur31** [comment]
So, that's not an NWBE instance running---it's the overall "hub controller". The idea (as I understand it) is that if this is up and running, NWBE instances will also run fine. (The MetaCell folks that manage our OSB infra asked us to use this URL as a test for uptime---I can ask them to explain this better).

Another thing to keep in mind is that not all NWB files on DANDI seem to work on NWBE for a multitude of reasons (one being that NWBE needs to be updated to handle newer NWB types). We did some validation that can be seen in the summary here:

https://github.com/OpenSourceBrain/DANDIArchiveShowcase/blob/main/validation_folder/README.md

Files marked there as "LL-P" and "LL-V" should generally work on NWBE.

**Comment by pgleeson** [comment]
@yarikoptic We're still getting pods for nwb-explorer starting every ~15 mins or so on our cluster, and unfortunately they're not getting killed successfully due to inactivity. Can you please switch off the uptime checks until we solve the issue with cleaning up pods on our cluster? It's causing the cluster to expand to the maximum number of nodes, costing us more.

**Comment by yarikoptic** [comment]
@pgleeson we never added uptime config for nwb explorer yet .  
Do you have IPs and agent strings for http requetss for  requests?  I thought that may be it is some crawler going through dandiarchive but unlikely since archive is JS driven and NWB explorer is behind some "dropdown" menu in the website interface -- so unlikely reachable by bots.
